crates/accelerate/src/commutation_analysis.rs

```diff
@@ -0,0 +1,179 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2024
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use pyo3::prelude::PyModule;
+use pyo3::{pyfunction, pymodule, wrap_pyfunction, Bound, PyResult, Python};
+use qiskit_circuit::operations::Param;
+use qiskit_circuit::Qubit;
+use smallvec::{smallvec, SmallVec};
+use std::hash::BuildHasherDefault;
+
+use crate::commutation_checker::CommutationChecker;
+use ahash::AHasher;
+use hashbrown::HashMap;
+use indexmap::IndexSet;
+use pyo3::prelude::*;
+
+use pyo3::types::{PyDict, PyList};
+use qiskit_circuit::dag_circuit::{DAGCircuit, NodeType, Wire};
+use rustworkx_core::petgraph::stable_graph::NodeIndex;
+
+type AIndexSet<T> = IndexSet<T, BuildHasherDefault<AHasher>>;
```

Comment:
This can be expressed a bit more succinctly with:

```suggestion
type AIndexSet<T> = IndexSet<T, ::ahash::RandomState>;
```

That being said I'm not sure I'm a fan of using a type alias for this instead of just using `IndexSet<NodeIndex, RandomState>` (assuming a use `ahash::RandomState` is added) where we need to define a new indexset.